lib/resources/user-preferences.js

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const Problem = require('../util/problem');
+const { success } = require('../util/http');
+
+module.exports = (service, endpoint) => {
+
+  //////////////////////////////////////////////////////////////////////////////
+  // User preferences (UI settings)
+
+  service.get('/user-preferences/current', endpoint(({ UserPreferences }, { auth }) =>
```

Comment:
I've been thinking about this URL for a bit. It feels like a lot? But I know it parallels `/users/current` to get basic account information about the logged in user. Because of that, I'm not sure if `/user-preferences` without `current` feels right because it _should_ just return info about the current user and not lead you to believe you can look up someone else's user preferences... 

Being able to set user preferences on the current user also makes it different. 

One idea (besides leaving it as is) would to just remove `current` and go with `/user-preferences` and have it documented to say it always works with the authenticated user only.

Another idea could be to piggy back off `/users/current`.
In the [docs](https://docs.getodk.org/central-api-accounts-and-users/#getting-authenticated-user-details), you can set extended metadata to get the current user's verbs, which is something _mostly_ meant for frontend. Originally, I thought "user preferences could be kept out of that, we'll add to the API, not change an existing endpoint". But I'm not sure the harm in just returning adding user preferences to the extended metadata _here_ on this existing endpoint. (It would make some things about the frontend simpler, too.) 

The only question then is what url to set the user preferences. PATCH to `/users/current/preferences`?

Reply:
I was thinking of `acteeId` being used in the backend only; it doesn't get exposed through the API but it could be useful generic database reference. The way I'm imagining it, you'd go from resource ID to actee ID to save the data and then actee ID back to resource ID to build the response data. But maybe that's an added layer of abstraction that doesn't need to be there and you could just be storing numeric project IDs like you're doing already.

From the API side of things, projects are referenced by their numeric ID and forms are usually referenced by projectId + xmlFormId. (We have numeric IDs for the forms, too, but we rarely expose those IDs.)

Can you think of a good URL pattern that would work for setting both site-wide and project preferences? 

`/user-preferences/key` and `/user-preferences/project/34/key`? 
`/user-preferences` and `/user-preferences?projectId=3` with `{key: value}` body?

Reply:
I'm in support of returning preferences in `/users/current` as part of the extended metadata!

Reply:
Yeah... good point! Let me convince myself this actee translation step is overly complex for what we're trying to accomplish:

We have a generic actee -> specific kind of object lookup happening in the audits query [here](https://github.com/getodk/central-backend/blob/master/lib/model/query/audits.js#L76-L100) so that audit events can describe specific kinds of projects/forms that they are attached to, but it's a complicated bit of code. We don't need something at that level of complexity for these user preferences for all the reasons you said - backend doesn't really care and it'll be more work to maintain. 

Is there something in between tracking `acteeIds` and the original generic approach?

`user_preferences` with columns `userId`, `projectId` (can be null), `propertyName`, `propertyValue`? 

Or do you want to go back to the original `userId` `body` approach? 

If you wanted to go back the original approach, could you add some tests showing setting/updating some preferences?

And if you want to include the user preferences in `/users/current` extended response, around [here](https://github.com/getodk/central-backend/blob/master/lib/resources/users.js#L111-L118) would be where to change the code.

Sorry for going back and forth on all these ideas, it's good we're discussing it!